server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -167,7 +162,7 @@ public boolean equals(Object o) {
             return false;
         }
 
-        return Arrays.equals(uncompressed(), that.uncompressed());
+        return uncompressed().equals(uncompressed());
```

Comment:
```suggestion
        return uncompressed().equals(that.uncompressed());
```

also looking at the code above, I wonder why we don't compare crc32 first before comparing the compressed byte arrays

Reply:
:facepalm: thanks for spotting.

>  I wonder why we don't compare crc32 first before comparing the compressed byte arrays

I guess you could argue that it's really unlikely that the compressed bytes in the equal (as in equal uncompressed bytes) case aren't actually equal so you'd just add an extra `int` comparison in the equal case. So if we assume it's mostly the equal case here then the current version is better, I have no clue if that's true though. Probably doesn't matter much in practice since byte array comparison in JDK9+ is blazing fast anyway via `jdk.internal.util.ArraysSupport#vectorizedMismatch`? :)

Reply:
> Probably doesn't matter much in practice

I agree that it doesn't matter much. Just a random thought that popped into my head :)